miraclelinux/meta-debian-extended
# Add logrotate

The logrotate version was different between Debian and poky. (Debian: 3.14.0, poky: 3.15.0)
Therefore, the patch file of poky is copied to meta-debian-extended.

・Overview
This is logrotate command test.
This test confirmed that the log file was rotated by logrotate command.

・test process
1. Create a file called test in the /etc/logrotate.d directory and set the rotation settings for the log to be rotated.
```
(File contents)
root@qemuarm64:/etc/logrotate.d# cat test
/var/log/test.log { ★Rotation target file name
create 0664 root root ★Setting: Creates a log file immediately after rotation.
maxsize 1M ★Setting: When the file exceeds 1MB, the rotation is executed regardless of the cron setting.
rotate 1 ★Setting: Number of files to rotate
}
```

2. Create a file called test.log in / var / log. (Content is appropriate, file size of 1MB or more)
```
root@qemuarm64:/etc/logrotate.d# cd /var/log/
root@qemuarm64:/var/volatile/log# cat /dev/urandom > test.log
^C
root@qemuarm64:/var/volatile/log# du sh test.log 
7.6M test.log
```

3. Test run logrotate and make sure there are no errors in the test file entries.
```
root@qemuarm64:/var/volatile/log# logrotate d /etc/logrotate.conf
(Excerpt)
rotating pattern: /var/log/test.log weekly (2 rotations)
empty log files are rotated, log files >= 1048576 are rotated earlier, old logs are removed
considering log /var/log/test.log
Now: 2019-09-06 05:33
Last rotated at 2019-09-06 05:31
log needs rotating
rotating log /var/log/test.log, log->rotateCount is 2
dateext suffix '-20190906'
glob pattern '[0-9][0-9][0-9][0-9][0-9][0-9][0-9][0-9]'
glob finding old rotated logs failed
moving /var/log/test.log to /var/log/test.log-20190906
creating new /var/log/test.log mode = 0664 uid = 0 gid = 0
```

4. Run logrotate and verify that the file is being rotated.
```
root@qemuarm64:/var/volatile/log# ls al
drwxr-xr-x 2 root root 140 Sep 6 05:25 .
drwxrwxrwt 4 root root 80 Sep 6 04:54 ..
-rw-r--r- 1 root root 0 Sep 6 04:54 boot
rw-r--r- 1 root root 15956 Sep 6 04:54 dmesg
rw-r--r- 1 root root 28358 Sep 6 05:26 messages
rw-r--r- 1 root root 7974912 Sep 6 05:25 test.log
rw-rw-r- 1 root root 58800 Sep 6 05:26 wtmp
root@qemuarm64:/var/volatile/log# logrotate /etc/logrotate.conf 
root@qemuarm64:/var/volatile/log# ls al
drwxr-xr-x 2 root root 160 Sep 6 05:28 .
drwxrwxrwt 4 root root 80 Sep 6 04:54 ..
-rw-r--r- 1 root root 0 Sep 6 04:54 boot
rw-r--r- 1 root root 15956 Sep 6 04:54 dmesg
rw-r--r- 1 root root 28358 Sep 6 05:26 messages
rw-rw-r- 1 root root 0 Sep 6 05:28 test.log ★File generated immediately after rotation
rw-r--r- 1 root root 7974912 Sep 6 05:25 test.log-20190906 ★Rotated file
rw-rw-r- 1 root root 58800 Sep 6 05:26 wtmp
```
